pkg/patterns/declarative/reconciler.go

```diff
@@ -154,6 +153,41 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, err
 	}
 
+	var clusters []addoncluster.Cluster
+	// TODO(yuwenma): Right now, the remote cluster control only support customized  `remoteClusterControl`,
+	// where users need to specify the Rest.Config for each cluster. We can also support a default version, which finds
+	// out the rest.config from .kubconfig based on the `spec.clusters[].name`.
+	if r.options.remoteClusterControl != nil {
+		remoteClusters, err := r.options.remoteClusterControl.GetClusters(ctx, instance)
+		if err != nil {
+			log.Error(err, "filtering remote cluster failed, not reconciling")
+		}
+		for _, remote := range remoteClusters {
+			clusters = append(clusters, addoncluster.NewRemote(ctx, remote.GetName(), remote.GetConfig(ctx)))
+		}
+	}
+	if len(clusters) == 0 {
+		// Local cluster
+		clusters = append(clusters, addoncluster.NewLocal(r.mgr))
+	}
+
+	objects, err := r.PrepareDeploymentObjects(ctx, request.NamespacedName, instance)
+	if err != nil {
+		log.Error(err, "unable to prepare deployment objects")
+		return reconcile.Result{}, err
+	}
+
+	for _, c := range clusters {
```

Comment:
todo: change to go routine. Does controller-runtime or client-go provide any utiles for iterations in `Reconclie`  method?